core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/GlobalUpdateExecutor.java

```diff
@@ -58,39 +61,49 @@ public GlobalUpdateExecutor(StorageManager storageManager, SonarLintWsClient wsC
 
-    progress.setProgressAndCheckCancel("Fetching global properties", 0.2f);
-    Set<String> allowedPlugins = globalPropertiesDownloader.fetchGlobalPropertiesTo(temp);
+    try {
+      progress.setProgressAndCheckCancel("Checking server version and status", 0.1f);
+      ServerInfos serverStatus = statusChecker.checkVersionAndStatus();
+      progress.setProgressAndCheckCancel("Checking plugins versions", 0.15f);
+      pluginsChecker.checkPlugins(serverStatus.getVersion());
 
-    progress.setProgressAndCheckCancel("Fetching plugins", 0.4f);
-    pluginReferenceDownloader.fetchPluginsTo(temp, allowedPlugins);
+      ProtobufUtil.writeToFile(serverStatus, temp.resolve(StorageManager.SERVER_INFO_PB));
 
-    progress.setProgressAndCheckCancel("Fetching rules", 0.6f);
-    rulesDownloader.fetchRulesTo(temp);
-    
-    progress.setProgressAndCheckCancel("Fetching list of modules", 0.8f);
-    moduleListDownloader.fetchModulesList(temp);
+      progress.setProgressAndCheckCancel("Fetching global properties", 0.2f);
+      Set<String> allowedPlugins = globalPropertiesDownloader.fetchGlobalPropertiesTo(temp);
 
-    progress.startNonCancelableSection();
-    progress.setProgressAndCheckCancel("Finalizing...", 1.0f);
-    
-    UpdateStatus updateStatus = UpdateStatus.newBuilder()
-      .setClientUserAgent(wsClient.getUserAgent())
-      .setSonarlintCoreVersion(VersionUtils.getLibraryVersion())
-      .setUpdateTimestamp(new Date().getTime())
-      .build();
-    ProtobufUtil.writeToFile(updateStatus, temp.resolve(StorageManager.UPDATE_STATUS_PB));
+      progress.setProgressAndCheckCancel("Fetching plugins", 0.4f);
+      pluginReferenceDownloader.fetchPluginsTo(temp, allowedPlugins);
 
-    Path dest = storageManager.getGlobalStorageRoot();
-    FileUtils.deleteDirectory(dest);
-    FileUtils.forceMkDirs(dest.getParent());
-    FileUtils.moveDir(temp, dest);
-  }
+      progress.setProgressAndCheckCancel("Fetching rules", 0.6f);
+      rulesDownloader.fetchRulesTo(temp);
+
+      progress.setProgressAndCheckCancel("Fetching list of modules", 0.8f);
+      moduleListDownloader.fetchModulesList(temp);
 
+      progress.startNonCancelableSection();
+      progress.setProgressAndCheckCancel("Finalizing...", 1.0f);
+
+      UpdateStatus updateStatus = UpdateStatus.newBuilder()
+        .setClientUserAgent(wsClient.getUserAgent())
+        .setSonarlintCoreVersion(VersionUtils.getLibraryVersion())
+        .setUpdateTimestamp(new Date().getTime())
+        .build();
+      ProtobufUtil.writeToFile(updateStatus, temp.resolve(StorageManager.UPDATE_STATUS_PB));
+
+      Path dest = storageManager.getGlobalStorageRoot();
+      FileUtils.deleteDirectory(dest);
+      FileUtils.forceMkDirs(dest.getParent());
+      FileUtils.moveDir(temp, dest);
+    } catch (RuntimeException e) {
+      try {
+        FileUtils.deleteDirectory(temp);
+      } catch (RuntimeException ignore) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)